masq/src/non_interactive_mode.rs

```diff
@@ -377,40 +384,91 @@ mod tests {
     }
 
     #[test]
-    fn populate_interactive_dependencies_produces_terminal_interface_blocking_printing_from_another_thread_when_the_lock_is_acquired(
+    fn populate_interactive_dependencies_produces_all_needed_to_block_printing_from_another_thread_when_the_lock_is_acquired(
     ) {
         let (test_stream_factory, test_stream_handle) = TestStreamFactory::new();
-        // This thread will leak, and will only stop when the tests stop running.
         let (broadcast_handle, terminal_interface) =
             Main::populate_interactive_dependencies(test_stream_factory).unwrap();
         {
-            let _lock = terminal_interface.lock();
-
+            let _lock = terminal_interface.as_ref().unwrap().lock();
             broadcast_handle.send(UiNewPasswordBroadcast {}.tmb(0));
+
             let output = test_stream_handle.stdout_so_far();
 
             assert_eq!(output, "")
         }
-
-        thread::sleep(Duration::from_millis(200)); //because of Win from Actions (theoretically others too)
+        //because of Win from Actions (theoretically some other platform too)
+        thread::sleep(Duration::from_millis(200));
 
         let output_when_unlocked = test_stream_handle.stdout_so_far();
+
         assert_eq!(
             output_when_unlocked,
             "\nThe Node\'s database password has changed.\n\n"
         )
     }
 
     #[test]
-    fn populate_interactive_dependencies_produces_a_functional_broadcast_handle() {
-        let (test_stream_factory, test_stream_handle) = TestStreamFactory::new();
-        // This thread will leak, and will only stop when the tests stop running.
-        let (broadcast_handle, _) =
-            Main::populate_interactive_dependencies(test_stream_factory).unwrap();
-        broadcast_handle.send(UiNewPasswordBroadcast {}.tmb(0));
+    fn extract_subcommands_can_process_interactive_mode_request() {
+        let args = vec!["masq".to_string()];
+
+        let result = Main::extract_subcommand(&args);
+
+        assert_eq!(result, None)
+    }
+
+    #[test]
+    fn extract_subcommands_can_process_non_interactive_request() {
+        let args = vec![
+            "masq".to_string(),
+            "setup".to_string(),
+            "--log-level".to_string(),
+            "off".to_string(),
+        ];
+
+        let result = Main::extract_subcommand(&args);
+
+        assert_eq!(
+            result,
+            Some(vec![
+                "setup".to_string(),
+                "--log-level".to_string(),
+                "off".to_string()
+            ])
+        )
+    }
+
+    #[test]
+    fn extract_subcommands_can_process_simple_help_requests() {
+        let args = vec!["masq".to_string(), "--help".to_string()];
+
+        let result = Main::extract_subcommand(&args);
+
+        assert_eq!(result, Some(vec!["--help".to_string()]))
```

Comment:
`--help` isn't a subcommand, it's a parameter. `setup` and `descriptor` and `set-wallets` are subcommands. If `extract_subcommand` extracts `--help`, it should be called something other than `extract_subcommand`. If the name continues to be `extract_subcommand`, then its operation should be adjusted so that it extracts only subcommands, and `--help` and `--version` should be handled elsewhere.